dvc/remote/gdrive/__init__.py

```diff
@@ -0,0 +1,246 @@
+            files_ids = self.cached_dirs[parts[0]]
+            parts.pop(0)
+
+        return files_ids, parents_ids, parts
+
+    def get_path_id(self, path_info, create=False):
+        files_ids, parents_ids, parts = self.get_path_id_from_cache(path_info)
+
+        if not parts and files_ids:
+            return files_ids[0]
+
+        file1 = self.resolve_remote_file(parents_ids, parts, create)
+        return file1["id"] if file1 else ""
+
+    def exists(self, path_info):
+        return self.get_path_id(path_info) != ""
+
+    def _upload(self, from_file, to_info, name, no_progress_bar):
+        dirname = to_info.parent
+        if dirname:
+            parent_id = self.get_path_id(dirname, True)
+        else:
+            parent_id = to_info.netloc
+
+        upload_request = RequestUploadFile(
+            {
+                "drive": self.drive,
+                "title": to_info.name,
+                "parent_id": parent_id,
+                "mime_type": "",
+            },
+            no_progress_bar,
+            from_file,
+            name,
+        )
+        self.execute_request(upload_request)
+
+    def _download(self, from_info, to_file, name, no_progress_bar):
+        file_id = self.get_path_id(from_info)
+        download_request = RequestDownloadFile(
+            {
+                "drive": self.drive,
+                "file_id": file_id,
+                "to_file": to_file,
+                "progress_name": name,
+                "no_progress_bar": no_progress_bar,
+            }
+        )
+        self.execute_request(download_request)
+
+    def list_cache_paths(self):
+        file_id = self.get_path_id(self.path_info)
+        prefix = self.path_info.path
+        for path in self.list_path(file_id):
+            yield posixpath.join(prefix, path)
+
+    def list_file_path(self, drive_file):
+        if drive_file["mimeType"] == FOLDER_MIME_TYPE:
+            for i in self.list_path(drive_file["id"]):
+                yield posixpath.join(drive_file["title"], i)
+        else:
+            yield drive_file["title"]
+
+    def list_path(self, parent_id):
+        for file1 in self.list_drive_item(
+            "'{}' in parents and trashed=false".format(parent_id)
+        ):
+            for path in self.list_file_path(file1):
+                yield path
+
+    def all(self):
+        if not self.cached_ids:
+            return
+
+        query = " or ".join(
+            "'{}' in parents".format(dir_id) for dir_id in self.cached_ids
+        )
+
+        query += " and trashed=false"
+        for file1 in self.list_drive_item(query):
+            parent_id = file1["parents"][0]["id"]
+            path = posixpath.join(self.cached_ids[parent_id], file1["title"])
+            try:
+                yield self.path_to_checksum(path)
+            except ValueError:
+                # We ignore all the non-cache looking files
+                LOGGER.debug('Ignoring path as "non-cache looking"')
```

Comment:
I think we use all lower case usually to name logger global var, please check the code around

Reply:
yep, the reason of uppercase was the warning on ci

Reply:
yep, take consistency with other code as a priority when there are doubts. Also, take deepsource and codeclimate as recommedation - no need to perfectly optimize for them - they are here to help us detect bugs and obvious anti-patterns